common/model-views.cpp

```diff
@@ -4183,6 +4234,133 @@ namespace rs2
+                        });
+
+                    }
+                    else
+                    {
+                        std::stringstream ss;
+                        ss << "Can not trigger Camera Accuracy Health, required streams not active";
+                        viewer.not_model->add_log(ss.str());
+                    }
+                }
+                else
+                {
+                    keep_showing = false;
+                }
+            }
+        }
+        break;
+
+        case camera_accuracy_health_model::model_state_type::PROCESS_MODAL:
+        {
+            bool process_finished(cah_model.calib_stts == RS2_CALIBRATION_SUCCESSFUL || cah_model.calib_stts == RS2_CALIBRATION_FAILED || cah_model.calib_stts == RS2_CALIBRATION_NOT_NEEDED);
+            
+            static std::map<rs2_calibration_status, std::string> status_map{
+                {RS2_CALIBRATION_SPECIAL_FRAME  , "In Progress" },
+                {RS2_CALIBRATION_STARTED        , "In Progress" },
+                {RS2_CALIBRATION_NOT_NEEDED     , "Ended" },
+                {RS2_CALIBRATION_SUCCESSFUL     , "Ended Successfully" },
+                {RS2_CALIBRATION_RETRY          , "In Progress" },
+                {RS2_CALIBRATION_FAILED         , "Ended With Failure" },
+                {RS2_CALIBRATION_SCENE_INVALID  , "In Progress" },
+                {RS2_CALIBRATION_BAD_RESULT     , "In Progress" } };
+
+
+            if (process_model("Camera Accuracy Health", "Camera Accuracy Health is In progress...  this may take a while.", status_map[cah_model.calib_stts], process_finished, window))
+            {
+                keep_showing = false;
+            }
+        }
+        break;
+        default:
+            break;
+
+        }
+
+        //reset internal elements
+        if (!keep_showing)
+        {
+            cah_model.cah_state = camera_accuracy_health_model::model_state_type::TRIGGER_MODAL;
+            cah_model.calib_stts = RS2_CALIBRATION_RETRY;
+        }
+        return keep_showing;
+        ;
+    }
+
+    bool device_model::prompt_reset_camera_accuracy_health(ux_window& window, std::string& error_message)
+    {
+        bool keep_showing = true;
+        bool yes_was_chosen = false;
+
+        std::string message_text("Would you like to reset Camera Accuracy Health ");
+        if (yes_no_dialog("Camera Accuracy Health", message_text, yes_was_chosen, window,error_message))
+        {
+            if (yes_was_chosen)
+            {
+                auto itr = std::find_if(subdevices.begin(), subdevices.end(), [](std::shared_ptr<subdevice_model> sub)
+                {
+                    if (sub->s->as<depth_sensor>())
+                        return true;
+                    return false;
+                });
+
+
+                if (itr != subdevices.end())
+                {
+                    auto sd = *itr;
+                    sd->s->set_option(RS2_OPTION_RESET_CAMERA_ACCURACY_HEALTH, 1.0f);
```

Comment:
We have other cameras that have depth sensors, but that do not have CAMERA_ACCURACY_HEALTH -- will this work?

Make sure you test it with a D400.

Reply:
It is only activated under this condition
                if (RS2_PRODUCT_LINE_L500 == parse_product_line(product_line_str))